src/interactions/User/AsMappingUser.js

```diff
@@ -11,15 +15,32 @@ export class AsMappingUser {
     Object.assign(this, user)
   }
 
+  allLayerSources() {
+    const customBasemaps = this.settings ?
+      _map(this.settings.customBasemaps,
+        basemap => basemapLayerSource(basemap, basemap.name)
+      ) : []
+
+    return _clone(LayerSources).concat(customBasemaps)
+  }
+
   defaultLayerSource() {
     if (!_isFinite(this.id) || !_isObject(this.settings)) {
       return null
     }
 
+    // Try to find a custom basemap if the user has one selected.
+    if (!_isEmpty(this.settings.defaultBasemapId)) {
+      return _find(this.allLayerSources(), (source) =>
```

Comment:
Any reason not to use the `findLayerSource` method you also added to this class?